src/components/Field.js

```diff
@@ -7,10 +7,70 @@ class Field extends React.Component {
         }
     }
 
+
+    fenNotation = () => {
+        
+        var positions = [];
+     
+        this.props.fields.map((field) => (

```

Comment:
Instead of using this.props.fields, try to get it through parameter and use return. Then make call of that method somewhere. Methods where everything going in through parameters and going out through return are more reusable and easier to test.